src/ucp/api/ucp.h

```diff
@@ -639,6 +678,33 @@ void ucp_context_print_info(ucp_context_h context, FILE *stream);
 ucs_status_t ucp_worker_create(ucp_context_h context, ucs_thread_mode_t thread_mode,
                                ucp_worker_h *worker_p);
 
+/**
+ * @ingroup UCP_WORKER
+ * @brief Create a worker object with list of parameters.
+ *
+ * This routine allocates and initializes a @ref ucp_worker_h "worker" object.
+ * Each worker is associated with one and only one @ref ucp_context_h
+ * "application" context.  In the same time, an application context can create
+ * multiple @ref ucp_worker_h "workers" in order to enable concurrent access to
+ * communication resources. For example, application can allocate a dedicated
+ * worker for each application thread, where every worker can be progressed
+ * independently of others.
+ *
+ * @note The worker object is allocated within context of the calling thread
```

Comment:
after talking with @shssf  - need to mark move ucp_worker_create() as deprecated - see ucp_ep_destroy() as an example of how to do it.